resources/builder.inc.sh

```diff
@@ -169,35 +169,73 @@ function builder_heading() {
 
 
 builder_echo() {
-  local color=white message= mark=
-  if [[ $# -gt 1 && $1 =~ ^(white|grey|green|success|blue|heading|yellow|warning|red|error|purple|brightwhite|teal|debug|setmark)$ ]]; then
-    color="$1"
-    shift
+  local color=white message= mark= block= action= do_output=true is_running_on_teamcity=false
+  if [[ -n "${TEAMCITY_GIT_PATH:-}" ]]; then
+    is_running_on_teamcity=true
+  fi
```

Comment:
Can we turn this into a function `builder_is_running_on_teamcity()` please, for future use and consolidation? We already have various places we are testing this, in different ways!